README.md

```diff
@@ -125,43 +127,40 @@ Then copy the `id` property value. It will be used in the next step as `SUBSCRIP
 
 ### 4. Create Variable Groups
 
-Get you service principal, your ACR credentials, and fill the following empty variables. Then, run this block on Bash:
+Ensure below steps are complete before creating variable groups on Azure Devops:
+- Azure Service Connection name - Use the service principal from above to set up service connection
+- Azure Key Vault name and set up below secrets within the vault with keys specified below:
+  - Set up the ARM client secret with key `arm-client-secret` assigned
+  - Set up the container registry secret with key `acr-secret` assigned
+- Azure Container Registry name
+
+Now fill the following empty variables below and run this block on Bash:
 
 ```shell
 CLIENT_ID=
-CLIENT_SECRET=
 TENANT_ID=
 SUBSCRIPTION_ID=
 ACR_NAME=
-ACR_PASSWORD=
+KEY_VAULT_NAME=
+SERVICE_CONNECTION_NAME=
 ```
 
 > Note: Make sure the `ACR_NAME` doesn't contain any capital letter, as it's an invalid ACR name convention.
 
-
-Then run the following commands to create the variable groups `JMETER_AZURE_PRINCIPAL` and `JMETER_TERRAFORM_SETTINGS`:
+Then run the following commands to create the variable group `JMETER_TERRAFORM_SETTINGS`:
 
 ```shell
-PRIN_GROUP_ID=$(az pipelines variable-group create  --name JMETER_AZURE_PRINCIPAL --authorize \
-                                                    --variables ARM_CLIENT_ID=$CLIENT_ID \
-                                                                ARM_TENANT_ID=$TENANT_ID \
-                                                                ARM_SUBSCRIPTION_ID=$SUBSCRIPTION_ID \
-                                                                | jq .id)
-
-az pipelines variable-group variable create --group-id $PRIN_GROUP_ID --secret true \
-                                            --name ARM_CLIENT_SECRET \
-                                            --value $CLIENT_SECRET
 
 SETT_GROUP_ID=$(az pipelines variable-group create  --name JMETER_TERRAFORM_SETTINGS --authorize \
                                                     --variables TF_VAR_JMETER_IMAGE_REGISTRY_NAME=$ACR_NAME \
                                                                 TF_VAR_JMETER_IMAGE_REGISTRY_USERNAME=$ACR_NAME \
                                                                 TF_VAR_JMETER_IMAGE_REGISTRY_SERVER=$ACR_NAME.azurecr.io \
+                                                                TF_VAR_SERVICE_CONNECTION_NAME=$SERVICE_CONNECTION_NAME \

```

Comment:
Service connection names autogenerated by AzDO usually have spaces. Spaces can break the evaluation of this statement, we can add quotes to it:
```shell
...
TF_VAR_SERVICE_CONNECTION_NAME="$SERVICE_CONNECTION_NAME" \
...
```

Reply:
changed